### PR TITLE
Prevent init from calling success and error. Add unit test coverage.

### PR DESCRIPTION
**Pull request by cianclarke**
Previously, our unit tests in `FHTests.m` never tested anything. They included a mock HTTP client, so avoided a good chunk of the SDK, and since we moved to blocks, the assertions contained within the block never impacted the outcome of the test - we were effectively testing nothing.  
This adds Nocilla, and tests a number of scenarios which we've come up against the past number of days which may not have had test coverage. Turns out there are no issues in the SDK, but now we have unit tests asserting this.  
There's huge room for improvement here, I think there's a lot we could expand upon - but right now, some tests are better than no tests!

### CODE REVIEW

File: fh-ios-sdk.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -10,7 +10,6 @@
 		48B4D0511B316CCC00385BAC /* libFeedHenry.a in Frameworks */ = {isa = PBXBuildFile; fileRef = B3D7C48315D28BE30019262E /* libFeedHenry.a */; };
 		48B4D0541B316F2500385BAC /* FHPushConfig.h in Headers */ = {isa = PBXBuildFile; fileRef = 48B4D0521B316F2500385BAC /* FHPushConfig.h */; };
 		48B4D0551B316F2500385BAC /* FHPushConfig.m in Sources */ = {isa = PBXBuildFile; fileRef = 48B4D0531B316F2500385BAC /* FHPushConfig.m */; };
-		55E51DF876A3F51961C76B14 /* libPods-FHTests.a in Frameworks */ = {isa = PBXBuildFile; fileRef = B07421CC258519C6ABDCAF54 /* libPods-FHTests.a */; };
```
Comment (cianclarke) [inline fh-ios-sdk.xcodeproj/project.pbxproj]: (I believe) Pod install alters this file and includes the correct file. This was causing build of the SDK to fail for me and others

File: fh-ios-sdk.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -54,7 +53,6 @@
 		6FCF31F81A9B430F0014B936 /* FHSyncNotificationMessage.h in Headers */ = {isa = PBXBuildFile; fileRef = 6F9C60031A9B366200C22C92 /* FHSyncNotificationMessage.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		6FCF31F91A9B43240014B936 /* NSString+MD5.h in Headers */ = {isa = PBXBuildFile; fileRef = 6F9C5FF21A9B366200C22C92 /* NSString+MD5.h */; settings = {ATTRIBUTES = (Private, ); }; };
 		6FCF31FA1A9B43240014B936 /* NSString+Validation.h in Headers */ = {isa = PBXBuildFile; fileRef = 6F9C5FF41A9B366200C22C92 /* NSString+Validation.h */; settings = {ATTRIBUTES = (Private, ); }; };
-		6FCF32071A9B47000014B936 /* FHAct+SetHttpClient.m in Sources */ = {isa = PBXBuildFile; fileRef = 6F9C5FC81A9B35F900C22C92 /* FHAct+SetHttpClient.m */; };
```
Comment (cianclarke) [inline fh-ios-sdk.xcodeproj/project.pbxproj]: no longer used

File: fh-ios-sdk.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -62,13 +60,13 @@
 		6FCF320C1A9B47000014B936 /* FHSyncUtilsTest.m in Sources */ = {isa = PBXBuildFile; fileRef = 6F9C5FD31A9B35F900C22C92 /* FHSyncUtilsTest.m */; };
 		6FCF320D1A9B47000014B936 /* FHTests.m in Sources */ = {isa = PBXBuildFile; fileRef = 6F9C5FD51A9B35F900C22C92 /* FHTests.m */; };
 		6FCF320E1A9B47000014B936 /* FHTestUtils.m in Sources */ = {isa = PBXBuildFile; fileRef = 6F9C5FD71A9B35F900C22C92 /* FHTestUtils.m */; };
-		6FCF320F1A9B47000014B936 /* MockFHHttpClient.m in Sources */ = {isa = PBXBuildFile; fileRef = 6F9C5FD91A9B35F900C22C92 /* MockFHHttpClient.m */; };
```
Comment (cianclarke) [inline fh-ios-sdk.xcodeproj/project.pbxproj]: no longer used

File: fh-ios-sdk.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -83,7 +81,7 @@
 		B3D7C55115D2923C0019262E /* CFNetwork.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = B3D7C55015D2923C0019262E /* CFNetwork.framework */; };
 		B3F685D81AD8352A002240DA /* FHDataManager.h in Headers */ = {isa = PBXBuildFile; fileRef = B3F685D61AD8352A002240DA /* FHDataManager.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		B3F685D91AD8352A002240DA /* FHDataManager.m in Sources */ = {isa = PBXBuildFile; fileRef = B3F685D71AD8352A002240DA /* FHDataManager.m */; };
-		E816D4D62D22C4E20B1F1233 /* libPods.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 2BC8A0A7417DA495FFDD185F /* libPods.a */; };
```
Comment (cianclarke) [inline fh-ios-sdk.xcodeproj/project.pbxproj]: As above note about `Pod install`

File: fh-ios-sdk.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -496,7 +486,7 @@
 				B3D7C48E15D28BE30019262E /* Sources */,
 				B3D7C48F15D28BE30019262E /* Frameworks */,
 				B3D7C49015D28BE30019262E /* Resources */,
-				B3D7C49115D28BE30019262E /* ShellScript */,
```
Comment (cianclarke) [inline fh-ios-sdk.xcodeproj/project.pbxproj]: ..maybe xCode updating automatically for 6.4, not 100% sure where this came from

File: fh-ios-sdk/FHHttpClient.m [hunk 6]
```diff
@@ -84,12 +84,8 @@ - (void)sendRequest:(FHAct *)fhact
         [fhResponse parseResponseData:responseData];
 
         if ([request responseStatusCode] == 200) {
-            NSString *status =
-                [fhResponse.parsedResponse valueForKey:@"status"];
-            if ((nil == status) || (nil != status)) {
```
Comment (cianclarke) [inline fh-ios-sdk/FHHttpClient.m]: this line does nothing. Remove our check for some arbitrary @"status" key

**Comment by cianclarke** [comment]
..new tests pass first time. Well, that was unexpected. travisCI && Cocoapods FTW

**Comment by corinnekrych** [comment]
+1 to merge it
Unit tests run well.
For mocking http layer we use OHHTTPStubs (but only available to mock NSURLSession) so Nocilla makes sense to mock ASIHttp layer.

**Comment by wei-lee** [comment]
@cianclarke This looks good. But which line is the fix to preventing init from calling success and error?

**Comment by cianclarke** [comment]
@corinnekrych Thanks for the review - yeah I really just picked most popular HTTP mocker I could find! 

@wei-lee Thanks for reviewing - here's the fix to prevent init and success from calling
https://github.com/feedhenry/fh-ios-sdk/pull/43/files#diff-8e87674fe8b32be696588c31def56447R88

Also this important fix re: problem we discussed yest:
https://github.com/feedhenry/fh-ios-sdk/pull/43/files#diff-8e87674fe8b32be696588c31def56447R87


Thanks again for jumping on this issue so quick @corinnekrych @wei-lee

**Pull request merged**